app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+class PostsController < ApplicationController
+  # include post_helper
+
+  before_action :logged_in_user, only: %i[create new]
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(post_params)
+    @user = current_user
+    if @user.posts.build(post_params).save
```

Comment:
The `13` and `15` lines it's the same.
One line it's sufficient

Reply:
We actually need the @post variable.
The reason for that is that when you submit a post that's not valid (empty body or title) we want the parts that were already written to still be in the #new view when it's rendered.
If there's no @post variable the render 'new' will  display an empty form regardless of what was first submited.